src/Command/PHPCRCleanupCommand.php

```diff
@@ -0,0 +1,336 @@
+        }
+    }
+
+    protected function configure()
+    {
+        $defaultDebugFile = \sprintf('%s/var/%s_phpcr-cleanup.md', $this->projectDirectory, \date('Y-m-d-H-i-s'));
+
+        $this->addOption('force', 'f', InputOption::VALUE_NONE, 'Do not ask for confirmation.');
+        $this->addOption('dry-run', null, InputOption::VALUE_NONE, 'Do not make any changes to the repository.');
+        $this->addOption('debug', null, InputOption::VALUE_NONE, 'Write debug information to a file.');
+        $this->addOption('debug-file', null, InputOption::VALUE_REQUIRED, 'Write debug information to a file.', $defaultDebugFile);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        $io->title('PHPCR Cleanup');
+
+        $dryRun = $input->getOption('dry-run');
+
+        if (!$dryRun) {
+            $io->warning('This command will remove properties from the PHPCR repository. Make sure to have a backup before running this command.');
+            if (!$input->getOption('force')) {
+                $answer = $io->ask('Do you want to continue [y/n]', null, function ($value) {
+                    $value = \strtolower($value);
+                    if (!\in_array($value, ['y', 'n'], true)) {
+                        throw new \RuntimeException('You need to enter "y" to continue or "n" to abort.');
+                    }
+
+                    return 'y' === $value;
+                });
+
+                if (!$answer) {
+                    $io->warning('You have aborted the command');
+
+                    return self::SUCCESS;
+                }
+            } else {
+                $io->writeln('The command will wait for 5 seconds before starting');
+                $progressBar = $io->createProgressBar(5);
+                $progressBar->start();
+                for ($i = 0; $i < 5; ++$i) {
+                    $progressBar->advance();
+                    \sleep(1);
+                }
+                $progressBar->finish();
+
+                $io->newLine();
+                $io->newLine();
+                $io->newLine();
+            }
+        }
+
+        $io->section('Initiating cleanup process ...');
+        $io->writeln('Project directory: ' . $this->projectDirectory);
+        $io->writeln('Dry-run: ' . ($dryRun ? 'enabled' : 'disabled'));
+
+        $debug = $input->getOption('debug');
+        $io->writeln('Debug: ' . ($debug ? 'enabled' : 'disabled'));
+        $this->logger = function (string $message) {
+            // do not print anything
+        };
+
+        if ($input->getOption('debug')) {
+            $debugFile = $input->getOption('debug-file');
+            $io->writeln('Debug file: ' . $debugFile);
+
+            $this->logger = function (string $message) use ($debugFile) {
+                \file_put_contents($debugFile, $message . \PHP_EOL, \FILE_APPEND);
+            };
```

Comment:
Why aren't we using OutputStreams here? Either use a `NullOutput` or a `StreamOutput` with a file handle. That way we don't have to keep opening the file over and over again. (And the usage syntax is nicer).

Reply:
makes totally sense - will change that